src/audio/volume/volume_hifi5.c

```diff
@@ -0,0 +1,516 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2024 Intel Corporation. All rights reserved.
+//
+// Author: Andrula Song <andrula.song@intel.com>
+
+/**
+ * \file
+ * \brief Volume HiFi5 processing implementation without peak volume detection
+ * \authors Andrula Song <andrula.song@intel.com>
+ */
+
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/common.h>
+#include <ipc/stream.h>
+#include <stddef.h>
+#include <stdint.h>
+
+LOG_MODULE_DECLARE(volume, CONFIG_SOF_LOG_LEVEL);
+
+#include "volume.h"
+
+#if SOF_USE_HIFI(5, VOLUME)
+
+#if (!CONFIG_COMP_PEAK_VOL)
+
+#include <xtensa/tie/xt_hifi5.h>
+
+/**
+ * \brief store volume gain 4 times for xtensa multi-way intrinsic operations.
+ * Simultaneous processing 4 samples.
+ * \param[in,out] cd Volume component private data.
+ * \param[in] channels_count Number of channels to process.
+ */
+static void vol_store_gain(struct vol_data *cd, const int channels_count)
+{
+	int32_t i;
+
+	/* using for loop instead of memcpy_s(), because for loop costs less cycles */
+	for (i = 0; i < channels_count; i++) {
+		cd->vol[i] = cd->volume[i];
+		cd->vol[i + channels_count * 1] = cd->volume[i];
+		cd->vol[i + channels_count * 2] = cd->volume[i];
+		cd->vol[i + channels_count * 3] = cd->volume[i];
+	}
+	cd->copy_gain = false;
+}
+
+#if CONFIG_FORMAT_S24LE
+/**
+ * \brief HiFi5 enabled volume processing from 24/32 bit to 24/32 or 32 bit.
+ * \param[in,out] dev Volume base component device.
+ * \param[in,out] sink Destination buffer.
+ * \param[in,out] source Input buffer.
+ * \param[in] frames Number of frames to process.
+ * \param[in] attenuation factor for peakmeter adjustment (unused)
+ */
+static void vol_s24_to_s24_s32(struct processing_module *mod, struct input_stream_buffer *bsource,
+			       struct output_stream_buffer *bsink, uint32_t frames,
+			       uint32_t attenuation)
+{
+	struct vol_data *cd = module_get_private_data(mod);
+	struct audio_stream *source = bsource->data;
+	struct audio_stream *sink = bsink->data;
+	ae_int32x2 in_sample, in_sample1;
+	ae_int32x2 out_sample, out_sample1;
+	ae_int32x2 volume, volume1;
+	ae_int32x4 *buf;
+	ae_int32x4 *buf_end;
+	int i, n, m;
+	ae_int32x4 *vol;
+	ae_valignx2 inu = AE_ZALIGN128();
+	ae_valignx2 outu = AE_ZALIGN128();
+	ae_int32x4 *in = (ae_int32x4 *)audio_stream_wrap(source,
+							 (char *)audio_stream_get_rptr(source)
+							 + bsource->consumed);
+	ae_int32x4 *out = (ae_int32x4 *)audio_stream_wrap(sink,
+							  (char *)audio_stream_get_wptr(sink)
+							  + bsink->size);
+	const int channels_count = audio_stream_get_channels(sink);
+	const int inc = sizeof(ae_int32x4);
+	int samples = channels_count * frames;
+
+	/** to ensure the adsress is 16-byte aligned and avoid risk of
```

Comment:
is this really supposed to be a doxygen comment? More below

Reply:
Fixed in next version